File: product-manager/backend/app/schemas/product.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class ProductBase(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    category: Optional[str] = None

class ProductCreate(ProductBase):
    pass

class ProductUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    price: Optional[float] = None
    category: Optional[str] = None

class ProductResponse(ProductBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: product-manager/frontend/src/types/product.ts
export interface Product {
    id: number;
    name: string;
    description: string;
    price: number;
    category: string;
    created_at: string;
    updated_at: string;
}
export interface ProductCreate {
    name: string;
    description: string;
    price: number;
    category: string;
}
export interface ProductUpdate {
    name?: string;
    description?: string;
    price?: number;
    category?: string;
}
